target.c

```diff
@@ -336,11 +336,14 @@ int tcmu_add_dev_to_recovery_list(struct tcmu_device *dev)
 		goto done;
 	}
 	list_add(&tpg_recovery_list, &tpg->recovery_entry);
+	rdev->flags |= TCMUR_DEV_FLAG_IN_RECOVERY;
+	rdev->conn_lost_cnt++;
```

Comment:
The increment should stay in `__tcmu_notify_conn_lost` IMO.  The point of my suggestion was to keep related things close and not mix them:
- set `TCMUR_DEV_FLAG_IN_RECOVERY` in the function that adds `rdev` to the respective list (using `rdev->recovery_entry`)
- increment `rdev->conn_lost_cnt` in the function that handles a lost connection

Reply:
Okay.